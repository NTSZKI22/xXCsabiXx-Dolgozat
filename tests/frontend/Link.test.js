import renderer from 'react-test-renderer';
import {render} from "@testing-library/react";
import Link from "../../resources/js/components/Link";

it('Link', () => {
    var component = renderer.create(
        <Link page="https://google.com">Google</Link>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props.onMouseEnter();
    })

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props.onMouseLeave();
    })
})

it('Link - #', () => {
    var component = renderer.create(
        <Link>Google</Link>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props.onMouseEnter();
    })

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props.onMouseLeave();
    })
})