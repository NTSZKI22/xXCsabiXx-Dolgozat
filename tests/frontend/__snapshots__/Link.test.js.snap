// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link - # 1`] = `
<a
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link - # 2`] = `
<a
  className="hovered"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link 1`] = `
<a
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link 2`] = `
<a
  className="hovered"
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;
